sys/vm_map.c

```diff
@@ -91,22 +94,31 @@ static bool vm_map_insert_entry(vm_map_t *vm_map, vm_map_entry_t *entry) {
 
 vm_map_entry_t *vm_map_find_entry(vm_map_t *vm_map, vm_addr_t vaddr) {
   vm_map_entry_t *etr_it;
-  TAILQ_FOREACH (etr_it, &vm_map->list, map_list)
-    if (etr_it->start <= vaddr && vaddr < etr_it->end)
+  rw_enter(&vm_map->rwlock, RW_READER);
+  TAILQ_FOREACH (etr_it, &vm_map->list, map_list) {
+    if (etr_it->start <= vaddr && vaddr < etr_it->end) {
+      rw_leave(&vm_map->rwlock);
```

Comment:
With many return points from function it's easier to forget to do unlock - it's probably better to call `rw_leave` in one place.